statemachine/statemachine.py

```diff
@@ -159,9 +162,35 @@ def allowed_events(self):
             for event in self.current_state.transitions.unique_events
         ]
 
-    def _process(self, trigger):
-        """This method will also handle execution queue"""
-        return trigger()
+    def _process(self, trigger):  # noqa: C901
+        # Check if the machine is running in "queued" mode
+        if not self._queued:
+            # If not, check if the queue is empty
+            if not self._external_queue:
+                # If the queue is empty, execute the trigger immediately
+                return trigger()
+            # If the queue is not empty, raise an exception as something is wrong
+            raise TransitionNotAllowed("Queue not empty.")
+
+        # If the machine is in "queued" mode, add the trigger to the queue
+        self._external_queue.append(trigger)
+        # If there is already a trigger in the queue,
+        # only return as the processing loop should already be running
+        if len(self._external_queue) > 1:
+            return
+
+        # Execute the triggers in the queue in order until the queue is empty
+        while self._external_queue:
+            # Get the next trigger in the queue
+            queued_trigger = self._external_queue[0]
+            try:
+                # Execute the trigger and remove it from the queue if successful
+                queued_trigger()
+                self._external_queue.popleft()
+            except Exception:
+                # If the trigger raises an exception, clear the queue and re-raise the exception
+                self._external_queue.clear()
```

Comment:
Here is also a kind of bold decision but what if there are a few triggers left in the queue that are totally fine? I guess you are right and we can't guarantee anything since the other triggers might depend on the one that failed.

Reply:
I don't know what's the right behavior and didn't find a reference. I think that Its a safe default and we should at least document It.